### PR TITLE
Decompile UpdateInput, GameLoop

### PR DESCRIPTION
**Pull request by mateon1**
And some other stuff.

### CODE REVIEW

**Review by SethBarberee — APPROVED** [review]
Looks pretty good aside from the one suggestion.

File: src/music.c [hunk 1]
```diff
@@ -183,7 +183,7 @@ void sub_800CB20(void)
 {
     sub_800BA5C();
     gUnknown_203B0B8++;
-    if((gUnknown_203B0BC & 5) != 0)
+    if((gRawKeyInput & 5) != 0)
```
[the remark this answers is not in the document]
Comment (mateon1) [reply src/music.c:186]: Right, I'll do that, I actually have that in my tree right now but I was working on another decomp.

File: src/pokemon.c [hunk 2]
```diff
@@ -7,7 +7,7 @@ extern u32 gUnknown_202F400;
 extern struct FileArchive gUnknown_8300500;
 extern const char gUnknown_81075F4;
 extern struct OpenedFile *gMonsterParametersFile;
-extern u16 gUnknown_20384E8;
+extern u16 gLevelCurrentPokeId;
```
[the remark this answers is not in the document]
Comment (mateon1) [reply src/pokemon.c:10]: This is the pokemon ID of the currently loaded Level table.

File: src/input.c [hunk 3]
```diff
@@ -1,73 +1,76 @@
...
 #include "input.h"
 
-EWRAM_DATA struct Input gUnknown_2025600;
-EWRAM_DATA struct UnkInputStruct1 gUnknown_20255F0;
-EWRAM_DATA struct UnkInputStruct1 gUnknown_2025638;
-EWRAM_DATA struct UnkInputStruct1 gUnknown_2025648;
-EWRAM_DATA struct UnkInputStruct2 gUnknown_2025668;
-EWRAM_DATA u32 gUnknown_202562C;
+EWRAM_DATA struct UnusedInputStruct gUnusedInputsRelated;
+EWRAM_DATA struct Inputs gRealInputs;
```
[the remark this answers is not in the document]
Comment (mateon1) [reply src/input.c:5]: - `ReadKeyInput` writes what it got directly to `gRawKeyInput`, which is used in music.c (which I suspect is an inaccurate name) for determining whether to make an extra RNG call.
- `UpdateInput` tracks two sets of inputs, the one from the last `UpdateInput` call, and the current one. It uses the `gLastInputs` for calculating things like 'button was pressed in this polling period', then ORs all inputs into a buffer `gBufferedInputs`.
- Functions in the game can then call `LoadBufferedInputs` on every non-lag frame (I think?), which moves the buffered inputs into `gRealInputs` and clears the buffer. The rest of the game only uses `gRealInputs`.

**Pull request merged**